### PR TITLE
feat: search history page

### PR DESCRIPTION
**Pull request by sshanzel**
**Note:** Don't update the branch yet.

## Changes
- Fixed a bug in the tabbed container for using a controlled state.
- Search history loading state.
- Search history empty state.
- Rendering Search history list (infinite scroll).

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1645 #done

### CODE REVIEW

File: packages/webapp/components/history/SearchEmpty.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { AiIcon } from '@dailydotdev/shared/src/components/icons';
+import { IconSize } from '@dailydotdev/shared/src/components/Icon';
+import { SearchHistoryContainer } from './common';
+
+export function SearchEmpty(): React.ReactElement {
```
Comment (rebelchris) [inline packages/webapp/components/history/SearchEmpty.tsx:6]: I think we always use just `ReactElement` doesn't really matter much, but it hit my OCD 😂

File: packages/webapp/components/history/search.tsx [hunk 2]
```diff
@@ -1,5 +1,60 @@
...
+      ({ pageParam }) => getSearchHistory({ after: pageParam, first: 30 }),
+      {
+        getNextPageParam: (lastPage) =>
+          lastPage?.history?.pageInfo.hasNextPage &&
+          lastPage?.history?.pageInfo.endCursor,
+      },
+    );
+
+  const canFetchMore =
+    !isLoading && !isFetchingNextPage && hasNextPage && data.pages.length > 0;
```
Comment (rebelchris) [inline packages/webapp/components/history/search.tsx:34]: Nothing to action on, but maybe web discussion, can we not add this to `useInifniteQuery` or wrap it, we always use the same

**Review by rebelchris — APPROVED** [review]
Looks good, some non breaking comments.

File: packages/webapp/components/history/SearchEmpty.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/webapp/components/history/SearchEmpty.tsx:6]: Lol. Yes, we should do it like that. Didn't notice this was not extracted.

File: packages/webapp/components/history/search.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/webapp/components/history/search.tsx:34]: I agree, we should probably create a hook to abstract these requirements. Kinda messy to copy paste it everywhere.

Comment (rebelchris) [reply packages/webapp/components/history/search.tsx:34]: Cool if it's easy feel free to do so, if not I can log it on the backlog

Comment (sshanzel) [reply packages/webapp/components/history/search.tsx:34]: Seems like there is more work involved relating to TypeScript. I think we can create a ticket and include the retro updates for existing implementations. Wdyt?

**Pull request merged**